syntax = "proto3";

package tesla.proto.energy_device.v1;

option go_package = "github.com/teslamotors/energy_device/pkg/protocol/protobuf/energydevice/v1";
option java_package = "com.tesla.generated.energy_device.v1";
option java_outer_classname = "AuthorizationMessages";

import "tesla_api/protobuf/energy_device/v1/authorization_api_add_authorized_client_by_trusted_signature_request.proto";
import "tesla_api/protobuf/energy_device/v1/authorization_api_add_authorized_client_by_trusted_signature_response.proto";
import "tesla_api/protobuf/energy_device/v1/authorization_api_add_authorized_client_request.proto";
import "tesla_api/protobuf/energy_device/v1/authorization_api_add_authorized_client_response.proto";
import "tesla_api/protobuf/energy_device/v1/authorization_api_configure_remote_service_request.proto";
import "tesla_api/protobuf/energy_device/v1/authorization_api_configure_remote_service_response.proto";
import "tesla_api/protobuf/energy_device/v1/authorization_api_get_signed_commands_public_key_request.proto";
import "tesla_api/protobuf/energy_device/v1/authorization_api_get_signed_commands_public_key_response.proto";
import "tesla_api/protobuf/energy_device/v1/authorization_api_list_authorized_clients_request.proto";
import "tesla_api/protobuf/energy_device/v1/authorization_api_list_authorized_clients_response.proto";
import "tesla_api/protobuf/energy_device/v1/authorization_api_remove_authorized_client_request.proto";
import "tesla_api/protobuf/energy_device/v1/authorization_api_remove_authorized_client_response.proto";

message AuthorizationMessages {
  oneof message {
    AuthorizationAPIAddAuthorizedClientRequest add_authorized_client_request                                          = 1;
    AuthorizationAPIAddAuthorizedClientResponse add_authorized_client_response                                        = 2;
    AuthorizationAPIRemoveAuthorizedClientRequest remove_authorized_client_request                                    = 3;
    AuthorizationAPIRemoveAuthorizedClientResponse remove_authorized_client_response                                  = 4;
    AuthorizationAPIListAuthorizedClientsRequest list_authorized_clients_request                                      = 5;
    AuthorizationAPIListAuthorizedClientsResponse list_authorized_clients_response                                    = 6;
    AuthorizationAPIGetSignedCommandsPublicKeyRequest get_signed_commands_public_key_request                          = 7;
    AuthorizationAPIGetSignedCommandsPublicKeyResponse get_signed_commands_public_key_response                        = 8;
    AuthorizationAPIAddAuthorizedClientByTrustedSignatureRequest add_authorized_client_by_trusted_signature_request   = 9;
    AuthorizationAPIAddAuthorizedClientByTrustedSignatureResponse add_authorized_client_by_trusted_signature_response = 10;
    AuthorizationAPIConfigureRemoteServiceRequest configure_remote_service_request                                    = 11;
    AuthorizationAPIConfigureRemoteServiceResponse configure_remote_service_response                                  = 12;
  }
}